examples/nextjs-spa/pages/recovery.tsx

```diff
@@ -0,0 +1,126 @@
+// React
+import { useEffect, useState } from "react"
+import { useRouter } from "next/router"
+
+// Ory SDK
+import { ory } from "../components/sdk"
+import { RecoveryFlow, UpdateRecoveryFlowBody } from "@ory/client"
+
+import { AxiosError } from "axios"
+import type { NextPage } from "next"
+
+import { UserAuthCard } from "@ory/elements"
+import React from "react"
+import Link from "next/link"
+
+const Recovery: NextPage = () => {
+  const [flow, setFlow] = useState<RecoveryFlow>()
+
+  // Get ?flow=... from the URL
+  const router = useRouter()
+  const { flow: flowId, return_to: returnTo } = router.query
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return
+    }
+
+    // If ?flow=.. was in the URL, we fetch it
+    if (flowId) {
+      ory
+        .getRecoveryFlow({ id: String(flowId) })
+        .then(({ data }) => {
+          setFlow(data)
+        })
+        .catch((err: AxiosError) => {
+          switch (err.response?.status) {
+            case 410:
+            // Status code 410 means the request has expired - so let's load a fresh flow!
+            case 403:
+              // Status code 403 implies some other issue (e.g. CSRF) - let's reload!
+              return router.push("/recovery")
+          }
+
+          throw err
+        })
+      return
+    }
+
+    // Otherwise we initialize it
+    ory
+      .createBrowserRecoveryFlow()
+      .then(({ data }) => {
+        setFlow(data)
+      })
+      .catch(
+        (error: AxiosError) =>
+          error.response?.status === 404 && router.push("/login"),
```

Comment:
why do we check for 404 if the URL should always resolve with the sdk? wouldn't redirecting to login just obscure the problem?

Reply:
couldn't we maybe redirect to the error page with a set message - something like:

```
the SDK threw a 404, this probably means the `NEXT_...ORY_SDK_URL` isn't set properly
```